### ISSUE TITLE
PR changing maintainers should notify original maintainers only

### ISSUE THREAD

**Issue by haampie** [spack/spack-bot#41]
https://github.com/spack/spack/pull/25156#issuecomment-890010384

The list of maintainers should be queried from the develop branch, not the PR itself.

**Comment by adamjstewart** [comment]
I disagree, the PR should ping both new and old maintainers. Consider this situation. Someone adds the developer of the software library as a maintainer of the recipe without their consent. We want them to know about this so they aren't being notified out of the blue on future PRs.

**Comment by alecbcs** [comment]
I think I agree with @adamjstewart it'd be jarring to have someone add you to a package as maintainer without your acknowledgment. @haampie do you agree with this assessment or are we overlooking something?
